packages/fetchai/skills/tac_participation/handlers.py

```diff
@@ -329,12 +330,48 @@ def _on_start(self, tac_message: TACMessage) -> None:
         game = cast(Game, self.context.game)
         game.init(tac_message, tac_message.counterparty)
         game.update_game_phase(Phase.GAME)
+        parameters = cast(Parameters, self.context.parameters)
+
+        if parameters.is_using_contract:
+            amount_by_currency_id = {
+                str(key): value
+                for key, value in tac_message.amount_by_currency_id.items()
+            }
+            quantities_by_good_id = {
+                str(key): value
+                for key, value in tac_message.quantities_by_good_id.items()
+            }
+            exchange_params_by_currency_id = {
+                str(key): value
+                for key, value in tac_message.exchange_params_by_currency_id.items()
+            }
+            utility_params_by_good_id = {
+                str(key): value
+                for key, value in tac_message.utility_params_by_good_id.items()
+            }
+
+            contract = self.context.contracts.erc1155
+            contract.set_deployed_instance(
+                self.context.ledger_apis.apis.get("ethereum"),
+                tac_message.get("contract_address"),
```

Comment:
contract address should be in the info dict imo

Reply:
The problem here is that you don't support info in the `GAME_DATA` message.